### ISSUE TITLE
Convergence warning

### ISSUE THREAD

**Issue by NicolasG93** [krlennon/mastercurves#4]
Hello,

I have an issue when superposing some creep data with mastercurves. I optimized the kernel and transform parameters but on some date I get the following error message : 
"C:\ProgramData\Anaconda3\lib\site-packages\sklearn\gaussian_process\_gpr.py:659: ConvergenceWarning: lbfgs failed to converge (status=2):
ABNORMAL_TERMINATION_IN_LNSRCH.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
  _check_optimize_result("lbfgs", opt_res)"

I tried to check how to increase maximum number of iterations in sklearn but I don't see how to implement this on the mastercurves module. I am supposed to modify "LogisticRegression()" by adding the argument "max_iter" with its new value.

I have to admit that I am quite new at coding with pyhton and I am a bit lost. Could you help me changing those parameters or solving this issue ?

Thanks in advance.

**Comment by krlennon** [comment]
Unfortunately, the optimizer behind the GPR is not exposed in the current version (scikit-learn doesn't allow you to set the parameters for its default optimizer, which is the reason this feature was left out). This is not an uncommon warning, so I'll plan on changing that in a future update.

This warning pops up in some of the demos, too, and doesn't necessarily mean that the underlying GPs used to construct the master curve are bad. Are you seeing unexpected/poor results from running your code? If that's the case, please provide a bit more detail to describe the issue.

**Comment by NicolasG93** [comment]
Thank you for your answer. Ok then I can't change those parameters. I get this message when the dataset I use are not very good (even by moving manually the curve, it is not easy to find a correlation between them). And when I use some better dataset, the warning doesn't appear and I find really correct results.

I was quite aware that the problematic dataset might be the cause of the convergence warning but I was wondering if there were adjustments to make to "force" the GPR to find a solution and build a mastercurve by changing for example the number of iteration made.

Thank you anyway for your answer.

**Comment by krlennon** [comment]
Yeah, if the self-similarity is not evident visually, it's unlikely that any automated superposition is going to work.

Worth noting though that this warning involves fitting the Gaussian Process models -- if it looks like the Gaussian processes have reasonably described the mean and variance in your data, and superposition still fails, then even increasing the number of iterations in the GPR routine wouldn't help.

Leaving this open until I have the chance to expose the optimizers for GPR. Feel free to comment if you have more questions/issues.

**Comment by slfuchs** [comment]
One can deactivate/ignore warnings from this package by adding the following lines to the python script that calls e.g. the method `mc.superpose()`:
```python
import warnings
warnings.filterwarnings('ignore', module='sklearn')
```
